### PR TITLE
Change: Croesus Chest Tracker can now determine chest that only can be open with a dungeon key

### PR DESCRIPTION
**Pull request by Thunderblade73**

### CODE REVIEW

**Review by hannibal002 — CHANGES_REQUESTED** [review]
Thanks for the changes!
Can you please add the missing empty lines in many places? (Especially around `if`, `else` and `{`)
If you use IntelliJ, you can use ⌘⌥L (macOS), or Ctrl+Alt+L (Windows/Linux)
See https://www.jetbrains.com/idea/guide/tips/reformat-file/

File: src/main/java/at/hannibal2/skyhanni/config/features/GUIConfig.java [hunk 1]
```diff
@@ -53,6 +53,12 @@ public static class TextBoxConfig {
     @FeatureToggle
     public boolean realTime = false;
 
+    @Expose
+    @ConfigOption(name = "Real Time 12h format", desc = "Display the current computer time in 12hr format rather than 24h format.")
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/config/features/GUIConfig.java]: Format should be capitalised

File: src/main/java/at/hannibal2/skyhanni/features/dungeon/CroesusUnopenedChestTracker.kt [hunk 2]
```diff
@@ -5,6 +5,7 @@ import at.hannibal2.skyhanni.events.GuiContainerEvent
 import at.hannibal2.skyhanni.utils.InventoryUtils
 import at.hannibal2.skyhanni.utils.ItemUtils.getLore
 import at.hannibal2.skyhanni.utils.LorenzColor
+import at.hannibal2.skyhanni.utils.LorenzDebug
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/features/dungeon/CroesusUnopenedChestTracker.kt]: Unused import

**Review by CalMWolfs — CHANGES_REQUESTED** [review]
looks good apart from the formatting, also please try to merge a separate branch in instead of from beta

**Comment by Thunderblade73** [comment]
Now it should be fine or did I miss something?

Also @CalMWolfs could you elarborate further on that:
> also please try to merge a separate branch in instead of from beta 

I didn't really understand this

**Comment by RayDeeUx** [comment]
> Now it should be fine or did I miss something?
> 
> Also @CalMWolfs could you elarborate further on that:
> 
> > also please try to merge a separate branch in instead of from beta
> 
> I didn't really understand this

hello yes, fellow contributor here who also learned of this the hard way

when you're faced with adding multiple features to skyhanni (or anything in general) it's better for everyone if you separated the additions by branches for a number of reasons.

for example, say i want to add an easter egg that sends people a smiley face whenever they earn an S+ score in M7, and also wanted to add an actual feature to prevent right clicking items with item abilities on private islands.

it would make more sense to have a pull request specifically for the easter egg and a second pull request for the item ability prevention feature, that way any changes on one pull request wouldn't directly affect changes from the other pull request.

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-branches in case you want the official lingo on pull requests

(@CalMWolfs is currently in bed at the moment as i type this, i hope this response is useful to some extent)

**Comment by Thunderblade73** [comment]
Ok, so everything that I implement should have it's own branch at my side and than an Pull Request to the main branch([hannibal002:beta](https://github.com/hannibal002/SkyHanni/tree/beta)). Did I understand this correctly?

**Comment by hannibal002** [comment]
> Ok, so everything that I implement should have it's own branch at my side and than an Pull Request to the main branch([hannibal002:beta](https://github.com/hannibal002/SkyHanni/tree/beta?rgh-link-date=2023-09-27T18%3A01%3A15Z)). Did I understand this correctly?


Yes, exactly like this!

**Review by hannibal002 — APPROVED** [review]

**Pull request merged**